wire/msgtx.go

```diff
@@ -701,11 +898,13 @@ func (msg *MsgTx) decodeWitness(r io.Reader, pver uint32, isFull bool) error {
 		}
 	}
 
+	writeTxInScriptsToMsgTx(msg, totalScriptSize)
```

Comment:
I think this needs a comment to explain what's going on since this would be quite unintuitive for a reader that is not already familiar with this code.